gpu/containers/include/pcl/gpu/containers/kernel_containers.h

```diff
@@ -36,73 +36,108 @@
-    {
-        template<typename T> struct DevPtr
-        {
-            using elem_type = T;
-            const static std::size_t elem_size = sizeof(elem_type);
-
-            T* data;
-
-            __PCL_GPU_HOST_DEVICE__ DevPtr() : data(nullptr) {}
-            __PCL_GPU_HOST_DEVICE__ DevPtr(T* data_arg) : data(data_arg) {}
-
-            __PCL_GPU_HOST_DEVICE__ std::size_t elemSize() const { return elem_size; }
-            __PCL_GPU_HOST_DEVICE__ operator       T*()       { return data; }
-            __PCL_GPU_HOST_DEVICE__ operator const T*() const { return data; }
-        };
-
-        template<typename T> struct PtrSz : public DevPtr<T>
-        {                     
-            __PCL_GPU_HOST_DEVICE__ PtrSz() : size(0) {}
-            __PCL_GPU_HOST_DEVICE__ PtrSz(T* data_arg, std::size_t size_arg) : DevPtr<T>(data_arg), size(size_arg) {}
-
-            std::size_t size;
-        };
-
-        template<typename T>  struct PtrStep : public DevPtr<T>
-        {   
-            __PCL_GPU_HOST_DEVICE__ PtrStep() : step(0) {}
-            __PCL_GPU_HOST_DEVICE__ PtrStep(T* data_arg, std::size_t step_arg) : DevPtr<T>(data_arg), step(step_arg) {}
-
-            /** \brief stride between two consecutive rows in bytes. Step is stored always and everywhere in bytes!!! */
-            std::size_t step;            
-
-            __PCL_GPU_HOST_DEVICE__       T* ptr(int y = 0)       { return (      T*)( (      char*)DevPtr<T>::data + y * step); }
-            __PCL_GPU_HOST_DEVICE__ const T* ptr(int y = 0) const { return (const T*)( (const char*)DevPtr<T>::data + y * step); }
-
-            __PCL_GPU_HOST_DEVICE__       T& operator()(int y, int x)       { return ptr(y)[x]; }
-            __PCL_GPU_HOST_DEVICE__ const T& operator()(int y, int x) const { return ptr(y)[x]; }
-        };
-
-        template <typename T> struct PtrStepSz : public PtrStep<T>
-        {   
-            __PCL_GPU_HOST_DEVICE__ PtrStepSz() : cols(0), rows(0) {}
-            __PCL_GPU_HOST_DEVICE__ PtrStepSz(int rows_arg, int cols_arg, T* data_arg, std::size_t step_arg) 
-                : PtrStep<T>(data_arg, step_arg), cols(cols_arg), rows(rows_arg) {}
-
-            int cols;
-            int rows;                                                                              
-        };
-    }
-
-    namespace device
-    {
-        using pcl::gpu::PtrSz;
-        using pcl::gpu::PtrStep;
-        using pcl::gpu::PtrStepSz;
-    }
-}
+namespace pcl {
+namespace gpu {
+template <typename T>
+struct DevPtr {
+  using elem_type = T;
+  const static std::size_t elem_size = sizeof(elem_type);
+
+  T* data;
+
+  __PCL_GPU_HOST_DEVICE__
+  DevPtr() : data(nullptr) {}
+
+  __PCL_GPU_HOST_DEVICE__
+  DevPtr(T* data_arg) : data(data_arg) {}
+
+  __PCL_GPU_HOST_DEVICE__ std::size_t
+  elemSize() const
+  {
+    return elem_size;
+  }
+
+  __PCL_GPU_HOST_DEVICE__
+  operator T*() { return data; }
+  __PCL_GPU_HOST_DEVICE__ operator const T*() const { return data; }
```

Comment:
Should this one be multiple lines as the one right above? and separated by a empty line?

Reply:
Thanks, @larshg! I corrected this.

Reply:
PS: You didn't push the change. Also, could you rebase on top of the master branch to resolve the conflict?

Reply:
I tried to rebase but run into some conflicts and would need to ask for advice. Should I checkout master and then type `git rebase NAME_OF_BRACH_FOR_THIS_PR` or checkout the branch for this PR and then type `git rebase master`. In both cases I get some merge conflicts. Should I then solve these conflicts? Sorry for these silly questions, but I'm learning how to deal with this and am grateful for some guidance!

Reply:
For rebase operations, I prefer using `rebase -i`. My workflow:
1. `git switch master` (or checkout instead of switch for older git)
2. `git pull <origin>`
3. `git switch -` (go back to previous branch)
4. `git rebase -i master`
5. Edit the file as needed (or just verify the operations about to happen)
6. Save and exit the file
7. Rebase is complete

Reply:
Thank you so much for your help, @kunaltyagi ! But I have the feeling that I did something wrong, there is way too much history in this PR now. There must have been a merge where it shouldn't have been. I'm terribly sorry. What can we do now?

Reply:
Ok, I have done another try and I think I managed to rebase the branch now as you explained to me, @kunaltyagi !  Sorry for the complication but I believe I begin to understand how rebase works. Nevertheless I did this in a different branch (gpu_clangformat_batch_second_try instead of the one we compare with here, gpu_clangformat_batch). Do you have any suggestions how we can deal with that?

Reply:
@larshg was so kind to explain me a few more steps about rebasing, so that I will give this another try. Again, I'm sorry for all these silly mistakes ;-)

Reply:
I think I could indeed rebase a local branch on my computer so that only the history and code relevant to this PR are visible. I wonder how I can push this to the remote so that it becomes publicly available and all the other (erroneously merged) commits are not part of this PR anymore. Should I force push the local branch? I understand that `git push -f` is a better option than `git push --force-with-lease` as I would want to ignore all these other changes that I have falsely merged. Is that correct?  Thanks to everybody in advance for their help!

Reply:
Just force push :)

BTW, force-with-lease is better than force while working with other people, but if it's just you (like this PR), both have same behavior

Reply:
Mamamia - all these commits that I erroneously merged are gone! Thank you so much, Lars and Kunal, for advising me on how to do that. Happy to learn that git became a little bit less intimidating.